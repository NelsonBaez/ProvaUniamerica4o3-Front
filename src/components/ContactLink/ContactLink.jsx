import React from 'react';
import {NavLink } from 'react-router-dom';


export default function ContactLink({contact}){
  const {id, name} = contact;

  return (
    <div >
      <NavLink 
        className={({ isActive }) => isActive ? "text-red-400" : "text-blue-400"}
        data-testid={`link-${id}`}
        id={`contato-${id}`}
        to={`/contatos/${id}`}>
        {name}
      </NavLink>
    </div>
  )
};