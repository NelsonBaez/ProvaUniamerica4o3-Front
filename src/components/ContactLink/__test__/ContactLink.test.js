import {render, screen, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ContactLink from '../ContactLink';
import {MemoryRouter} from 'react-router-dom';

test('should render contact links', () => {
  const contact = { id: 1, name: 'jorge', email: "jorge@teste", phone: "123123123" };
  render(
    <MemoryRouter><ContactLink contact={contact}/></MemoryRouter>
  )
  const element = screen.getByTestId('link-1');
  expect(element).toBeInTheDocument();
  expect(element).toHaveTextContent('jorge');

})

test('matches snapshot', () => {
  const contact = { id: 1, name: 'jorge', email: "jorge@teste", phone: "123123123" };
  const tree = renderer.create(
    
    <MemoryRouter><ContactLink contact={contact}/></MemoryRouter>
  
  ).toJSON();
  expect(tree).toMatchSnapshot();
  
});