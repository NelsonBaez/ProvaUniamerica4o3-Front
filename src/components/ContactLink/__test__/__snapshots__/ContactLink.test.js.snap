// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <a
    className="blue"
    data-testid="link-1"
    href="/contatos/1"
    onClick={[Function]}
  >
    jorge
  </a>
</div>
`;
